sympy/polys/polytools.py

```diff
@@ -6218,8 +6218,20 @@ def sqf_list(f, *gens, **args):
     (2, [(x + 1, 2), (x + 2, 3)])
 
     """
-    return _generic_factor_list(f, gens, args, method='sqf')
-
+    options.allowed_flags(args, ['frac', 'polys'])
+    opt = options.build_options(gens, args)
+    f = sympify(f)
+    if isinstance(f, Expr) and not f.is_Relational:
+        numer, denom = together(f).as_numer_denom()
+        numer = Poly(numer, *gens, **args)
+        if denom!=1 and not opt.frac:
+            raise PolynomialError("a polynomial expected, got %s" % f)
```

Comment:
These two lines can be removed.

Reply:
Okay, thanks your review. Will change these things